src/main/java/software/amazon/lambda/snapstart/ByteCodeIntrospector.java

```diff
@@ -42,7 +42,7 @@ private static Set<String> setOf(String ... strings) {
     };
 
     boolean isLambdaHandler(XClass xClass) {
-        return implementsLambdaInterface(xClass) || hasLambdaHandlerMethod(xClass);
+        return implementsLambdaInterface(xClass) || hasLambdaHandlerMethod(xClass) || isLambdaHandlerField(xClass);
```

Comment:
It's somewhat weird that `isLambdaHandler` includes `isLambdaHandlerField`.

In my mind it makes more sense if the check for `isLambdaHandlerField` is done during the `LambdaHandlerIninitedWithRandomValue` class, and here they just stay as two separate methods (Like how Crac resources are also checked separately).